network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/PhaseTapChangerImpl.java

```diff
@@ -148,4 +146,35 @@ public static void validateStep(TapChangerStepAttributes step, TapChangerParent
             throw new ValidationException(parent, "step alpha is not set");
         }
     }
+
+    // equals and hashcode are override because of ValidationUtil.checkPhaseTapChangerRegulation that
+    // remove the tap changer of the list of allRegulatingTapChanger
+    // and it does not work if equals and hascode check this.attributesGetter
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PhaseTapChangerImpl that = (PhaseTapChangerImpl) o;
+        // check phase tap changer are on same leg
+        if (!Objects.equals(that.getTransformer().getClass(), getTransformer().getClass())) {
+            return false;
+        }
+        if (that.getTransformer() instanceof ThreeWindingsTransformerImpl &&
+            !Objects.equals(((ThreeWindingsTransformerImpl.LegImpl) parent).getSide(),
+                ((ThreeWindingsTransformerImpl.LegImpl) that.getParent()).getSide())) {
+            return false;
+        }
+        return Objects.equals(getTransformer().getId(), that.getTransformer().getId()) &&
+            Objects.equals(getRegulationMode(), that.getRegulationMode()) &&
+            Objects.equals(getRegulationValue(), that.getRegulationValue());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getParent(), getTransformer().getId(), getRegulationMode(), getRegulationValue());
```

Comment:
getParent is used in equals to check if the parent is a leg of a threewindings transformer or or twowindings transformer

Reply:
well precisely we use parent.getSide()

Reply:
I can change Hashcode to parent.getSide too

Reply:
Maybe to complex, let's keep it like that

Reply:
> Maybe to complex, let's keep it like that

It would actually be better to have consistency between equals and hashcode, sorry for previous comment!